src/extras/core/ShapePath.js

```diff
@@ -13,6 +13,7 @@ function ShapePath() {
 	this.type = 'ShapePath';
 
 	this.color = new Color();
+	this.opacity = 1.0;
```

Comment:
The OP has added it as a feature ("Add global opacity support for Shapes."). But we decided to  leave all presentation to the user code. After all, the opacity can be obtained by ```path.style.opacity```. Appearance properties should be obtained by the user always from the path style, to not bloat the loader code.

Reply:
Yeah, it's on purpose here - I agree most props from SVG affect only SVGs and can be accessed via `style`, but `opacity` is kind of a direct part of the 3D Shape rendering (I just didn't want to touch the Color class and add `alpha` support there - in case Color's used in other places.)

As in this example: https://threejs.org/docs/#examples/en/loaders/SVGLoader , nobody really would ever need `fillStyle` or `strokeWidth` to create a Material, but it's intuitive to use:

```js
var material = new THREE.MeshBasicMaterial( {
  color: path.color,
  opacity: path.opacity,
  transparent: true,
  side: THREE.DoubleSide,
  depthWrite: false
} );
```

But it's up to your judgement - should I remove it?

Reply:
Yes, it should be:
```
var material = new THREE.MeshBasicMaterial( {
  color: path.color,
  opacity: path.style.opacity !== undefined ? path.style.opacity : 1,
  transparent: path.style.opacity !== undefined ? ( path.style.opacity < 1 ) : false,
  side: THREE.DoubleSide,
  depthWrite: false
} );
```
I know it is not as handy, but it is the only way to not start bloating the loader code with style properties.

Reply:
Gotcha, done.
Went through [SVG attributes](https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/Presentation#attr-color-profile) to see if there's anything else that may be missing, added `visibility`. Rest seems unsupported / unsupportable with ShapeBuffers?
Also, is there a reason why we couldn't add `<text>` support (in another PR)? There's already [TextGeometry](https://threejs.org/docs/#api/en/geometries/TextGeometry) around. 🤔

Reply:
Thanks!